featuretools/feature_base/features_deserializer.py

```diff
@@ -73,13 +81,22 @@ def __init__(self, features_dict):
         self._primitives_deserializer = PrimitivesDeserializer()
 
     @classmethod
-    def load(cls, features):
+    def load(cls, features, profile_name):
         if isinstance(features, str):
             try:
                 features_dict = json.loads(features)
             except ValueError:
-                with open(features, 'r') as f:
-                    features_dict = json.load(f)
+                transport_params = {}
+                session = boto3.Session()
+                if isinstance(profile_name, str):
+                    transport_params = {'session': boto3.Session(profile_name=profile_name)}
+                if _is_s3(features) and (session.get_credentials() is None or profile_name is False):
```

Comment:
If the user doesn't have a default profile but does have a named profile in their credentials file this logic will end up using the anonymous way of reading the file.